core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -851,6 +902,11 @@ protected Builder buildAggregate(Aggregate e, Builder builder,
    * @return true if selectList is required to be added in sqlNode
    */
   boolean isStarInAggregateRel(Aggregate e) {
+
+    PivotRelTrait relTrait = e.getTraitSet().getTrait(PivotRelTraitDef.instance);
+    if (relTrait != null && relTrait.getsubqueryPresentFlag()) {
+      return true;
+    }
     if (e.getAggCallList().size() > 0) {
```

Comment:
simplify the condition like this: `!e.getAggCallList().isEmpty()`

Reply:
resolved